src/components/policy/policy_regular/src/cache_manager.cc

```diff
@@ -1217,13 +1224,19 @@ bool CacheManager::SetPredataPolicy(const std::string& app_id) {
   return true;
 }
 
-bool CacheManager::IsPredataPolicy(const std::string& app_id) {
+bool CacheManager::IsPredataPolicy(const std::string& app_id) const {
   // TODO(AOleynik): Maybe change for comparison with pre_DataConsent
   // permissions or check string value from get_string()
-  policy_table::ApplicationParams& pre_data_app =
-      pt_->policy_table.app_policies_section.apps[kPreDataConsentId];
-  policy_table::ApplicationParams& specific_app =
-      pt_->policy_table.app_policies_section.apps[app_id];
+  if (!IsApplicationRepresented(app_id)) {
+    return false;
+  }
+
+  policy_table::ApplicationPolicies& apps =
```

Comment:
@AByzhynar The use of `operator []` requires that `apps` should not be `const`. The same answer applies like the one for the previous question.